src/Microsoft.ApplicationInsights/TelemetryClient.cs

```diff
@@ -516,44 +518,95 @@ public void Initialize(ITelemetry telemetry)
+                            {
+                                telemetryWithSampling.IsProactivelySampledOut = true;
+                                sampledOut = true;
+                                break;
+                            }
+                            else
+                            {
+                                shouldCheckSamplingScore = false;
+                            }
+                        }
+
+                        try
+                        {
+                            this.configuration.TelemetryInitializers[index].Initialize(telemetry);
+                        }
+                        catch (Exception exception)
+                        {
+                            CoreEventSource.Log.LogError(string.Format(
+                                                            CultureInfo.InvariantCulture,
+                                                            "Exception while initializing {0}, exception message - {1}",
+                                                            this.configuration.TelemetryInitializers[index].GetType().FullName,
+                                                            exception));
+                        }
+                    }                    
                 }
-                catch (Exception exception)
+                else
                 {
-                    CoreEventSource.Log.LogError(string.Format(
-                                                    CultureInfo.InvariantCulture,
-                                                    "Exception while initializing {0}, exception message - {1}",
-                                                    initializer.GetType().FullName,
-                                                    exception));
+                    for (int index = 0; index < this.configuration.TelemetryInitializers.Count; index++)
+                    {
+                        try
+                        {
+                            this.configuration.TelemetryInitializers[index].Initialize(telemetry);
+                        }
+                        catch (Exception exception)
+                        {
+                            CoreEventSource.Log.LogError(string.Format(
+                                                            CultureInfo.InvariantCulture,
+                                                            "Exception while initializing {0}, exception message - {1}",
+                                                            this.configuration.TelemetryInitializers[index].GetType().FullName,
+                                                            exception));
+                        }
+                    }
                 }
             }
 
-            if (telemetry.Timestamp == default(DateTimeOffset))
+            if (!sampledOut)
```

Comment:
if (sampledOut) - then log that rest of Initialize is skipped.

Reply:
purely for investigation purposes if something behaves unexpectedly.. like we are seeing telemetry with unknown timestamp or sdkversion etc, and and to isolate it has nothing to do with the new proactive sampling.